knotx-server/src/main/java/io/knotx/server/KnotxGatewayContextHandler.java

```diff
@@ -61,6 +66,12 @@ private void handleRoute(final RoutingContext context, final String address) {
     if (StringUtils.isNotBlank(bodyAsString)) {
       knotContext.setFragments(Collections.singletonList(Fragment.raw(bodyAsString)));
     }
+    Set<FileUpload> fileUploads = context.fileUploads();
+    if (fileUploads != null) {
+      knotContext.setFilesData(fileUploads.stream().map(fileUpload -> {
+        return new FileData(fileUpload);
+      }).collect(Collectors.toList()));
```

Comment:
What's the reason we're switching between `Set` and `List`? Is this a limitation of an existing API or is `List` simply more suitable for the `knotContext` domain?